### PR TITLE
Closing JarFile instances

### PR DESCRIPTION
**Pull request by kiroco12**
In the Eval class, we are creating instances of `JarFile` which are not garbaged correctly. This only calls close when we obtain the data we want.

### CODE REVIEW

File: server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala [hunk 1]
```diff
@@ -463,14 +463,21 @@ class ${className} extends (() => Any) with java.io.Serializable {
...
+                            val nestedClassPath = Try {
+                              val jar = new JarFile(jarFile)
+                              val CP  = jar.getManifest.getMainAttributes.getValue("Class-Path")
+                              jar.close()
+                              CP
+                            } match {
+                              case Success(classPath) => classPath
+                              case Failure(throwable) =>
+                                throw new CompilerException(List(List(throwable.getMessage)))
+                            }
```
Comment (fedefernandez) [inline server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: Taking into account that we're not implementing this with effects, I'd go for a safer code:

```scala
  val nestedClassPath: String = try {
    val jar = new JarFile("")
    try {
      jar.getManifest.getMainAttributes.getValue("Class-Path")
    } finally {
      try {
        jar.close()
      } catch {
        case NonFatal(_) =>
      }
    }
  } catch {
    case NonFatal(e) =>
      throw new CompilerException(List(Option(e.getMessage).toList))
  }
```

File: server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala [hunk 2]
```diff
@@ -462,28 +462,35 @@ class ${className} extends (() => Any) with java.io.Serializable {
...
+        val relativeRoot = new File(jarFile).getParentFile
+        val jarResource: Resource[IO, JarFile] =
+          Resource.make(IO(new JarFile(jarFile)))(jar => IO(jar.close()))
+
+        val nestedClassPath: IO[String] =
+          jarResource
+            .use(jar => IO(jar.getManifest.getMainAttributes.getValue("Class-Path")))
+            .handleError { throwable: Throwable =>
+              throw new CompilerException(List(List(throwable.getMessage)))
+            }
```
Comment (juanpedromoreno) [inline server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: Following Fede's suggestion after my PR:

```scala
            .handleError { case scala.util.control.NonFatal(e) =>
              throw new CompilerException(List(List(e.getMessage)))
            }
```

Comment (kiroco12) [reply server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: I have a question here, what is doing here the `scala.util.control.NonFatal(e)`?

Comment (juanpedromoreno) [reply server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: https://www.scala-lang.org/api/2.12.10/scala/util/control/NonFatal$.html

Comment (fedefernandez) [reply server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala]: I'm almost sure is not necessary for the cats `ApplicativeError` methods (they already avoid handling fatal exceptions) but it's ok to have it.

**Pull request merged**